revocation/revocation.go

```diff
@@ -0,0 +1,122 @@
+// Package Revocation provides methods for checking the revocation status of a certificate chain
+package revocation
+
+import (
+	"crypto/x509"
+	"net/http"
+	"time"
+
+	"github.com/notaryproject/notation-core-go/revocation/ocsp"
+)
+
+// Revocation is an interface that specifies methods used for revocation checking
+type Revocation interface {
+	Validate(certChain []*x509.Certificate, signingTime time.Time) error
+	OCSPStatus(certChain []*x509.Certificate, signingTime time.Time) error
+	SetMergeErrorsFunction(mergeErrorsFunc func(errors []error) error)
```

Comment:
What are your thoughts on the previous suggestions of returning the list of errors instead of a merged error and providing a separate function that would merge the errors to determine if it was OK, Unknown, or Revoked?